backend/src/org/commcare/session/SessionNavigator.java

```diff
@@ -95,36 +96,38 @@ private void handleGetDatum() {
 
     private void handleAutoSelect() {
         SessionDatum selectDatum = currentSession.getNeededDatum();
-        if (selectDatum.getLongDetail() == null) {
-            // No confirm detail defined for this entity select, so just set the case id right away
-            // and proceed
-            String autoSelectedCaseId = SessionDatum.getCaseIdFromReference(
-                    currentAutoSelectedCase, selectDatum, ec);
-            currentSession.setDatum(selectDatum.getDataId(), autoSelectedCaseId);
-            startNextSessionStep();
-        } else {
-            sendResponse(LAUNCH_CONFIRM_DETAIL);
+        if (selectDatum instanceof EntityDatum) {
+            EntityDatum entityDatum = (EntityDatum)selectDatum;
```

Comment:
The check and cast to `EntityDatum` are unfortunate, but at least it is nice that we are being explicit that this behavior is only supported for case selections and not other datums.